chain/chain/src/resharding.rs

```diff
@@ -317,25 +324,49 @@ impl Chain {
         let checked_account_id_to_shard_uid =
             get_checked_account_id_to_shard_uid_fn(shard_uid, new_shards, next_epoch_shard_layout);
 
+        let shard_uid_string = shard_uid.to_string();
+        let metrics_labels = [shard_uid_string.as_str()];
+
         // Once we build the iterator, we break it into batches using the get_trie_update_batch function.
-        while let Some(batch) = get_trie_update_batch(&mut iter) {
+        loop {
```

Comment:
In here each of the three phases is wrapped in a dedicated scope with a relevant timer. 
Slightly easier to review with whitespace hidden.